ui/shared/JobInfo.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getInspectTaskUrl, getJobSearchStrHref } from '../helpers/url';
+import { getSearchStr, getTimeFields } from '../helpers/job';
+
+export default class JobInfo extends React.PureComponent {
```

Comment:
PureComponent only does a shallow comparison. If job metadata changes when the JobInfo component has already been rendered (eg an already selected job), would the changes to the `job` prop be reflected?

Reply:
This is a good point.  But being a ``PureComponent`` here is actually still good because we SHOULD get a new ``selectedJob`` instance.  The problem is, we're not.  So I actually fixed that in PR #4215.